frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.tsx

```diff
@@ -116,10 +120,6 @@ export function ObjectDetailView({
     if (table && _.isEmpty(table.fks) && !isVirtualCardId(table.id)) {
       fetchTableFks(table.id as ConcreteTableId);
     }
-    // load up FK references
-    if (!_.isEmpty(tableForeignKeys)) {
-      loadFKReferences();
-    }
```

Comment:
The first time Object Detail Modal is mounted, this condition did not execute.
On subsequent mounts, this `loadFKReferences` call was duplicating what `useEffect` takes care of later on.